api/v1/helpers.go

```diff
@@ -47,3 +67,70 @@ func (l LocalSpec) ToVolumeAndMount(volName string) (core.Volume, core.VolumeMou
 	}
 	return vol, mnt
 }
+
+// GetBucketAndPrefix return bucket and the prefix used in the backend
+func (backend Backend) GetBucketAndPrefix() (string, string, error) {
+	if backend.Local != nil {
+		return "", filepath.Join(backend.Local.MountPath, strings.TrimPrefix(backend.Local.SubPath, "/")), nil
+	} else if backend.S3 != nil {
+		return backend.S3.Bucket, strings.TrimPrefix(backend.S3.Prefix, backend.S3.Bucket+"/"), nil
```

Comment:
`strings.TrimPrefix(backend.S3.Prefix, backend.S3.Bucket+"/")`

Why are we doing this?

Reply:
This is for older version of Stash. In `v1alpha1` api, we used to set entire path inside bucket as prefix of Repostiory. For S3 the path includes bucket name as prefix. Hence, to get actual prefix we have remove the bucket name.